### PR TITLE
[Login] Create login screen

### LINKED ISSUE #9: [Login] Create login screen
## Overview (Required)

- Create login view

## Todo

- [x] Create login view
- [x] Create login business logic

### PR DESCRIPTION
**Pull request by wsi1212**
## Overview (Required)

- Create login view

## Screenshot

 After 
  

## Issue
- close #9

### CODE REVIEW

**Review by SeokgyuYun — CHANGES_REQUESTED** [review]

**Comment by Mercen-Lee** [comment]
LGTM ^^

File: build-logic/src/main/kotlin/com/b1nd/dodam/primitive/HiltPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            with(pluginManager) {
+                apply("com.google.devtools.ksp")
+            }
+            dependencies {
+                implementation(libs.library("hilt-core"))
+                ksp(libs.library("hilt-compiler"))
+            }
+        }
+    }
+
```
Comment (Mercen-Lee) [inline build-logic/src/main/kotlin/com/b1nd/dodam/primitive/HiltPlugin.kt:23]: Why blank line??

**Review by Mercen-Lee — CHANGES_REQUESTED** [review]

File: settings.gradle.kts [hunk 2]
```diff
@@ -21,7 +21,13 @@ include(
...
     ":network:core",
+    ":network:login",
     ":network:meal",
+    ":data:login",
+    ":model",
     ":common",
-)
\ No newline at end of file
+)
+include(":keystore")
```
Comment (SeokgyuYun) [inline settings.gradle.kts:32]: It would be better to write this include code as above

**Comment by Sanghyun0505** [comment]
LGBT 🏳️‍🌈

File: feature/login/src/main/java/com/b1nd/dodam/login/viewmodel/LoginViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        loginRepository.login(id, encryptSHA512(pw)).collect { token ->
+            when (token) {
+                is Result.Success -> {
+                    datastoreRepository.saveUser(
+                        id = id,
+                        pw = pw,
+                        token = token.data.accessToken,
+
+                    )
+                    datastoreRepository.user.collect {
```
Comment (SeokgyuYun) [inline feature/login/src/main/java/com/b1nd/dodam/login/viewmodel/LoginViewModel.kt:34]: Remove this logging code please

**Review by Mercen-Lee — APPROVED** [review]
LGTM 👍

File: feature/login/src/main/java/com/b1nd/dodam/login/navigation/LoginNavigation.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import androidx.navigation.NavController
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavOptions
+import androidx.navigation.compose.composable
+import com.b1nd.dodam.login.LoginScreen
+
+const val LOGIN_ROUTE = "login"
+
+fun NavController.navigationToLogin(navOptions: NavOptions?) = navigate(LOGIN_ROUTE, navOptions)
```
Comment (SeokgyuYun) [inline feature/login/src/main/java/com/b1nd/dodam/login/navigation/LoginNavigation.kt:11]: Add default value `null` to navOptions please

File: feature/login/src/main/java/com/b1nd/dodam/login/LoginScreen.kt [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.b1nd.dodam.designsystem.component.DodamFullWidthButton
+import com.b1nd.dodam.designsystem.component.DodamTextField
+import com.b1nd.dodam.designsystem.theme.BackIcon
+import com.b1nd.dodam.designsystem.theme.DodamTheme
+import com.b1nd.dodam.login.viewmodel.Event
+import com.b1nd.dodam.login.viewmodel.LoginViewModel
+
+@Composable
+internal fun LoginScreen(viewModel: LoginViewModel = hiltViewModel(), onBackClick: () -> Unit, navigateToMain: () -> Unit) {
```
Comment (SeokgyuYun) [inline feature/login/src/main/java/com/b1nd/dodam/login/LoginScreen.kt:30]: It would be better to put parameters on searate lines

**Review by SeokgyuYun — CHANGES_REQUESTED** [review]

File: network/login/src/main/java/com/b1nd/dodam/network/login/api/LoginService.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import com.b1nd.dodam.network.login.model.LoginRequest
+import com.b1nd.dodam.network.login.model.LoginResponse
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import javax.inject.Inject
+internal class LoginService @Inject constructor(
```
Comment (Mercen-Lee) [inline network/login/src/main/java/com/b1nd/dodam/network/login/api/LoginService.kt:16]: Why no line break here?

**Review by Mercen-Lee — CHANGES_REQUESTED** [review]

**Review by SeokgyuYun — APPROVED** [review]
LGTM 👍

**Review by Mercen-Lee — APPROVED** [review]
LGTM ^^

**Pull request merged**